_posts/2023-11-08-routines.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "Share your best Trino SQL routine"
+author: "Martin Traverso, Dain Sunstrom, David Phillips, Manfred Moser"
+excerpt_separator: <!--more-->
+image: /assets/trino-sql-routine.png
+show_pagenav: false
+---
+
+We want to see the best [SQL routines]({{site_url}}/docs/current/routines.html)
+you can write, feature them as [examples in the
+documentation]({{site_url}}/docs/current/routines/examples.html), and send you
+some goodies as a reward!
+
+<!--more-->
+
+With the recent [Trino 431
+release]({{site_url}}/docs/current/release/release-431.html) we shipped a
+feature that has been awaited by many Trino users for a long, long time. [SQL
+routines]({{site_url}}/docs/current/routines.html) are an easy way to define our
+own procedural, custom functions. All users on your Trino instance can then use
+that function in their queries and enjoy the new feature to simplify their
+queries.
+
+The new process of writing a routine in your client tool in SQL replaces the old
+way of having to create a custom plugin in Java, compiling it, and getting the
+binary deployed in your cluster. The time it takes to use a function has gone
+from hours to minutes!
+
+Our documentation includes details for all the supported statements:
+
+* `BEGIN`
+* `CASE`
+* `DECLARE`
+* `FUNCTION`
+* `IF`
+* `ITERATE`
+* `LEAVE`
+* `LOOP`
+* `REPEAT`
+* `RETURN`
+* `SET`
+* `WHILE`
+
+With the memory connector and the Hive connector supporting routine storage, you
+can use `CREATE FUNCTION` and `DROP FUNCTION`, so that everyone using the
+cluster has access to your routines.
+
+The unit tests and our [examples
+documentation](({{site_url}}/docs/current/routines/examples.html)) contain a
+number of routines that scratch the surface of what is possible. Now, we are
+looking for you to help us improve the documentation and maybe even find some
+bugs. So here is what we are asking from you:
+
+* Send us your best SQL routine.
+  * Create a pull request to add to the [examples in the
+    documentation](https://github.com/trinodb/trino/blob/master/docs/src/main/sphinx/routines/examples.md)
+    with a new section, and request a review from [Manfred
+    (mosabua)](https://github.com/mosabua)
+  * Alternatively, [email the details](mailto:manfred@starburst.io) and submit a
+    [CLA](https://github.com/trinodb/cla) separately.
+* Explain the use case, what the routine does, and maybe also how it works.
+* Include the full statement for the `CREATE FUNCTION` definition and an example
+  invocation.
+* Add any necessary tables or data so we can test the function.
+
```

Comment:
Maybe add a sentence saying if you need help writing a function, ask on some channel on slack

Reply:
Added